apps/mochi-web/components/app/detail/AppDetailIntegration.tsx

```diff
@@ -60,17 +74,63 @@ export const AppDetailIntegration = ({ apiKey = '' }: Props) => {
             Secret key
           </Typography>
           <div className="flex my-2 space-x-2">
-            <div className="rounded border border-neutral-300 shadow-input px-3.5 flex items-center flex-1">
-              <Typography level="p5">********* ******** ********</Typography>
+            <div className="rounded border border-neutral-300 shadow-input px-3.5 flex items-center flex-1 overflow-hidden">
+              <Typography
+                level="p5"
+                className="text-ellipsis whitespace-nowrap"
+              >
+                {secretKey || '********* ******** ********'}
+              </Typography>
             </div>
-            <Button variant="outline" color="neutral" className="!bg-neutral-0">
-              Reset
-            </Button>
+            {secretKey ? (
+              <>
+                <Tooltip
+                  content="Copied"
+                  arrow="top-center"
+                  componentProps={{
+                    root: { open: hasCopiedSecretKey },
+                    trigger: { asChild: true },
+                  }}
+                >
+                  <Button variant="outline" onClick={onCopySecretKey}>
+                    Copy
+                  </Button>
+                </Tooltip>
+                <Button
+                  variant="outline"
+                  color="neutral"
+                  className="!bg-neutral-0"
+                  onClick={onResetSecretKey}
+                >
+                  Generate new
+                </Button>
+              </>
+            ) : (
+              <Button
+                variant="outline"
+                color="neutral"
+                className="!bg-neutral-0"
+                onClick={onResetSecretKey}
```

Comment:
Although it occurs quickly, could we implement a disable state for the reset/generate button?